docs/benchmark/ftlib_benchmark.md

```diff
@@ -0,0 +1,171 @@
+worker resource to do experiments. And, we also use CIFAR-10 dataset and
+assign more CPUs and memory for the worker. The experiment parameters are:
+
+| Experiment parameter | Value |
+| -- | --- |
+| Batch size | 64 |
+| Batches per task | 50 |
+| Images per task | 3220 |
+| Dataset | CIFAR-10 |
+| Input shape | (32, 32, 3) |
+| Output classes| 10 |
+| Worker resource| cpu=4,memory=8192Mi,ephemeral-storage=1024Mi|
+
+### ResNet50 with CIFAR-10 on an On-premise CPU Cluster
+
+| Workers   | communication   |  Speed   |  Speedup Ratio |
+| -- | --------------------------- | -------- | ------ |
+| 1  |   0%  | 26.7 images/s  |  1  |
+| 2  | 18% | 41 images/s | 1.57 |
+| 4  | 25% | 68.4 images/s  | 2.56 |
+| 8  | 32% | 123 images/s | 4.61 |
+
+### MobileNetV2 with CIFAR-10 on an On-premise CPU Cluster
+
+| Workers   | communication   |  Speed   |  Speedup Ratio |
+| -- | --------------------------- | -------- | ------ |
+| 1  |   0%  | 353.6 images/s  |  1  |
+| 2  | 24% | 503 images/s | 1.42 |
+| 4  | 44.7% | 680 images/s  | 1.92 |
+| 8  | 66.7% | 648 images/s  | 1.83 |
+
+For the above experiments of CPU, we find that the speed-up ratio is better if
+the model is more complex. Because the computation on CPUs is
+very slow. The computation is the bottleneck on the CPU cluster. So, the
+the speed-up of ResNet50 is better than MobileNetV2.
+
+## On-premise GPU Cluster Managed by a Tailored Kubernetes System
+
+On the on-premise GPU cluster, we can launch pods with single P100 GPU.
+As we known, GPUs are faster for computing than CPUs and the size of
+CIFAR-10 is small to test the performance with GPUs. We use ImageNet with
+input shape (256,256,3) to train Resnet50, MobileNetV2, and an iamge
+compression model. The last model has less parameters tensors.
+
+Experiment parameters are:
+
+| Experiment parameter | Value    |
+| -- | --- |
+| Batch size | 64 |
+| Batches per task | 16 |
+| Images per task | 1024 |
+| Dataset | ImageNet |
+| Image shape | (256, 256, 3) |
+| Output classes| 100 |
+| Worker resource| cpu=8,gpu=1,memory=16000Mi,ephemeral-storage=1024Mi|
+
+### ResNet50 with ImageNet on an On-premise GPU Cluster
+
+There are also 214 trainable tensors in ResNet50 for ImageNet. But the
+model trainable parameters is 23,739,492 because there are 100 classes.
+
+| Workers   | speed | total task time  | allreduce time| tensor.numpy() time| apply_gradients |
+| --------- | ----- | --------------- | -------- | ------ | ---------- |
+| 1 (local) | 168 images/s| 6.1s  |  - | - |  4.16s |
+| 2  | 148 images/s| 13.76s | 10.36s | 5.04s | 1.35s |
+| 4  | 228 images/s| 18s | 14.67s |  5.14s | 1.30s |
+
+![Resnet50](https://user-images.githubusercontent.com/18071380/88752262-0fdd2800-d18c-11ea-8b37-83ba5662e53d.png)
+
+### MobileNetV2 with ImageNet on an On-premise GPU Cluster
+
+For ImageNet dataset with 100 classes, there are also 158 trainable tensors in
+MobileNetV2 for ImageNet. And those tensors totally have 2,386,084 trainable parameters.
```

Comment:
So, the parameters memory size is  2386084 * 4 bytes？A float32 parameter occupies 4 bytes.

2386084 * 4 bytes is about 9.1MB

Reply:
Yes. It is about 9.1MB